onvm/onvm_mgr/onvm_stats.c

```diff
@@ -222,37 +229,89 @@ onvm_stats_clear_nf(uint16_t id) {
 }
 
 void
-onvm_stats_add_event(const char *msg, struct onvm_nf_info *nf_info) {
-        if (msg == NULL || stats_destination != ONVM_STATS_WEB) {
+gen_event_info(const char *msg, uint8_t type, void *data) {
+        struct onvm_event *event;
+
+        event = (struct onvm_event *)malloc(sizeof(struct onvm_event));
+        if (event == NULL) {
+                perror("Couldn't allocate event");
+                return;
+        }
+        
+        event->type = type;
+        event->msg = msg;
+        event->data = data;
+
+        onvm_stats_add_event(event);
+}
+
+void
+gen_event_nf_info(const char *msg, struct onvm_nf_info *nf_info) {
+        struct onvm_event *event;
+
+        event = (struct onvm_event *)malloc(sizeof(struct onvm_event));
+        if (event == NULL) {
+                perror("Couldn't allocate event");
+                return;
+        }
+
+        event->type = 4;
+        event->msg = msg;
+        event->data = nf_info;
+
+        onvm_stats_add_event(event);
+}
+
+/****************************Internal functions*******************************/
+
+static void
+onvm_stats_add_event(struct onvm_event *event_info) {
+        if (event_info == NULL || stats_destination != ONVM_STATS_WEB) {
                 return;
         }
         char event_time_buf[20];
-        time_t time_raw_format;
+        uint8_t type;
         struct tm *ptr_time;
+        struct onvm_nf_info *nf_info;
+        time_t time_raw_format;
         time(&time_raw_format);
+        type = event_info->type;
+
         ptr_time = localtime(&time_raw_format);
         if (strftime(event_time_buf, 20, "%F %T", ptr_time) == 0) {
                 perror("Couldn't prepare formatted string");
         }
-        cJSON *new_event = cJSON_CreateObject();
+
         cJSON *source = cJSON_CreateObject();
+        cJSON *new_event = cJSON_CreateObject();
         cJSON_AddStringToObject(new_event, "timestamp", event_time_buf);
-        cJSON_AddStringToObject(new_event, "message", msg);
-        if (nf_info == NULL) {
-                cJSON_AddStringToObject(source, "type", "MGR");
-        } else {
-                cJSON_AddStringToObject(source, "type", "NF");
+        cJSON_AddStringToObject(new_event, "message", event_info->msg);
+
+        if (type == 0) {
```

Comment:
Skimmed through this is the part that needs to be changed, avoid magic numbers. Always do `#define COOL_MSG_NAME 2`